src/app/campaign/store/reducers/campaign.reducer.ts

```diff
@@ -21,115 +21,92 @@ export const initialState: CampaignState = {
   saving: false
 };
 
-export function reducer(state = initialState, action: CampaignActions | AdvertiserActions): CampaignState {
-  switch (action.type) {
-    case ActionTypes.CAMPAIGN_NEW: {
-      return { ...initialState, loading: false, loaded: true };
-    }
-    case ActionTypes.CAMPAIGN_DUP_FROM_FORM: {
-      return {
-        localCampaign: duplicateCampaign(action.payload.campaign),
-        changed: true,
-        valid: true,
-        loading: false,
-        loaded: true,
-        saving: false
-      };
-    }
-    case ActionTypes.CAMPAIGN_DUP_BY_ID_SUCCESS: {
-      return {
-        localCampaign: duplicateCampaign(docToCampaign(action.payload.campaignDoc)),
-        changed: true,
-        valid: true,
-        loading: false,
-        loaded: true,
-        saving: false
-      };
-    }
-    case ActionTypes.CAMPAIGN_FORM_UPDATE: {
-      const { campaign, changed, valid } = action.payload;
-      return {
-        ...state,
-        localCampaign: {
-          ...state.localCampaign,
-          ...campaign
-        },
-        changed,
-        valid
-      };
-    }
-    case ActionTypes.CAMPAIGN_ADD_ADVERTISER_SUCCESS: {
-      const { set_advertiser_uri } = docToAdvertiser(action.payload.doc);
```

Comment:
`createReducer` uses `on` with action reducer functions instead of a switch
* it has built in type safety with action creators
* doesn't need a default case
* at some point, they also added support for processing an action type in more than one `on` so that if an action type is seen more than once, the reducer is wrapped and the additional `on` is applied. (not used in this app, just interesting and different than what you can do with switch)